source/MissionAction.cpp

```diff
@@ -176,6 +176,13 @@ void MissionAction::Load(const DataNode &node, const string &missionName)
 			// Create a GameData reference to this mission name.
 			GameData::Missions().Get(toFail);
 		}
+		else if(key == "system")
+		{
+			if(system.empty() && child.HasChildren())
```

Comment:
Generally unless there's a good reason not to, I allow data specifications to break a particular data entry into multiple pieces. So, I don't think there's a need to check `system.empty()` here.

Reply:
`system` is the `on enter` MissionAction's named system, e.g. for `on enter Sol` system is `Sol`

This check is to prevent someone from explicitly naming a system at the root level of the MissionAction, and *also* defining a LocationFilter to pick the system.